### PR TITLE
Update ITK to post-5.3rc03 including requiring at least C++14

### PR DESCRIPTION
**Pull request by jamesobutler**
This is part of work for #5673.

C++14 is the new minimum with CMake 3.16.3 as set by ITK 5.3 requirements (https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3rc01/Documentation/SupportedCompilers.md).

This will enable future GrowCut remote ITK module work as @dzenanz was beginning in #5807.

### CODE REVIEW

**Comment by jamesobutler** [comment]
No build issues on the Windows platform now. Tests that failed are consistent with already failing tests on the factory build machine (https://slicer.cdash.org/viewTest.php?onlyfailed&buildid=2456606).
```
1>The following tests FAILED:
1>	  6 - cmake_slicer_generate_extension_description_test (Failed)
1>	  8 - py_cmake_slicer_extensions_index_build_with_upload (Failed)
1>	  9 - py_cmake_slicer_extensions_index_build_with_upload_using_ctest (Failed)
1>	268 - qSlicerExtensionsManagerModelTest (Failed)
1>	280 - qSlicerMouseModeToolBarTest1 (Failed)
1>	332 - vtkMRMLAnnotationLinesNodeTest1 (Failed)
1>	334 - vtkMRMLAnnotationLinesStorageNodeTest1 (Failed)
```

**Review by dzenanz — DISMISSED** [review]

**Comment by jamesobutler** [comment]
Waiting for actual ITK 5.3 release and corresponding SimpleITK based off official release. Currently some issues with SimpleITK running latest ITK pre official ITK 5.3 release.

**Comment by jamesobutler** [comment]
@blowekamp It seems the SimpleITK support for ITK v5.3rc03 isn't quite working yet even though using commit https://github.com/SimpleITK/SimpleITK/commit/a885bba17147f7fb023d8f6cdf533a30b34ce5f6.
simpleitk-build-error

File: Docs/developer_guide/build_instructions/windows.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## Install prerequisites
 
-- [CMake](http://www.cmake.org/cmake/resources/software.html) version >= 3.21.1 (or 3.15.1 <= version < 3.20.4)
+- [CMake](http://www.cmake.org/cmake/resources/software.html) version that meets at least the minimum required CMake version [here](https://github.com/Slicer/Slicer/blob/master/CMakeLists.txt#L1). Avoid CMake 3.21.0 specifically.
```
Comment (jamesobutler) [inline Docs/developer_guide/build_instructions/windows.md:5]: @lassoan Here is where it is updated in the build instructions. I have it pointing directly to the place in the file so it wouldn’t get out of sync. 

However thinking about how the documentation gets tagged with a release, linking to `master` here could lead to issues when the `master` branch doesn’t match the stable version doc.

I’ll probably have to change this line to a specific text version and not have this type of link.

**Comment by jamesobutler** [comment]
Previously mentioned SimpleITK build issues have been resolved.

**Comment by cpinter** [comment]
Mentioning SimpleITK windows build issues, I just tested the very latest Slicer with a custom app (https://github.com/Slicer/Slicer/commit/015fa17a5715863052cda5dae2a14098589b01ae) and I get the following error:

```
  -- SimpleITK: E:/t3/tR/python-install/bin/PythonSlicer.exe;setup.py;install
  CMake Error at E:/t3/tR/slicersources-src/CMake/ExternalProjectForNonCMakeProject.cmake:104 (message):
    SimpleITK: install step failed with exit code '1'. 
```
In `SimpleITK_install_step_error.txt`:
```
Traceback (most recent call last):
  File "E:\t3\tR\SimpleITK-build\SimpleITK-build\Wrapping\Python\setup.py", line 14, in <module>
    from SimpleITK._version import __version__
  File "E:\t3\tR\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK\__init__.py", line 18, in <module>
    from SimpleITK.SimpleITK import *
  File "E:\t3\tR\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK\SimpleITK.py", line 13, in <module>
    from . import _SimpleITK
ImportError: DLL load failed while importing _SimpleITK: The specified procedure could not be found.
```
Could this be related? Any suggestions? Thank you!

**Comment by jamesobutler** [comment]
When Slicer was recently upgraded to a newer Python version in https://github.com/Slicer/Slicer/commit/34e48e8aef5dad19ec8a955d6f48a1940846e3f3, it required follow-up commit https://github.com/Slicer/Slicer/commit/a1c44fab014a6be93e5917fe63ffee3a98dba597 to fix a SimpleITK build issue. Is your custom app including this bug fix?

Using current Slicer `master` I don't believe I have seen SimpleITK build issues. The SimpleITK build issues mentioned in https://github.com/Slicer/Slicer/pull/5954#issuecomment-1020162123 were specifically from using latest ITK and latest SimpleITK without setting `-DSimpleITK_INT64_PIXELIDS:BOOL=ON`.

**Comment by cpinter** [comment]
Yes, as I wrote the custom app is on the very latest Slicer, hash linked in my previous comment.

**Comment by jamesobutler** [comment]
It does look like a non-clean build issue where SimpleITK wasn't built cleanly. As I remember potentially that exact same issue when building as of https://github.com/Slicer/Slicer/commit/34e48e8aef5dad19ec8a955d6f48a1940846e3f3. However, if you say it was a clean build, then I'm not sure. I haven't seen such issues on my own builds with current latest.

**Comment by cpinter** [comment]
With an incremental build the first error was almost immediate, so I'm 100% sure I did a clean build. Do you have any clue I could potentially follow?

**Comment by jamesobutler** [comment]
@fbordignon reported in https://github.com/SimpleITK/SimpleITK/issues/1513 appears to be the exact same build error you faced. He resolved that in https://github.com/Slicer/Slicer/pull/6026#issuecomment-970246699. That change was not included in https://github.com/Slicer/Slicer/commit/34e48e8aef5dad19ec8a955d6f48a1940846e3f3, but ultimately was included in https://github.com/Slicer/Slicer/commit/a1c44fab014a6be93e5917fe63ffee3a98dba597 as also observed in the commit message body.

**Comment by fbordignon** [comment]
Actually @jcfr fixed the issue in a more elegant way with python [site ](https://docs.python.org/3/library/site.html) package. Take a look into the final integration commit https://github.com/Slicer/Slicer/commit/34e48e8aef5dad19ec8a955d6f48a1940846e3f3#diff-8a0835d08a9451302636735091ab1593fb967cd685a3e18f91ead8a908eb4b3d

**Comment by cpinter** [comment]
Thanks both!

So @fbordignon you managed to build a SlicerCAT project with SimpleITK on?

I double and triple checked that these changes are included and yes they are, but the build failed nevertheless. I don't know which paths are the ones that need to be added so can't check that, but I'll run one more clean build to triple check that too.

**Comment by fbordignon** [comment]
> So @fbordignon you managed to build a SlicerCAT project with SimpleITK on?

Yes, I was using my solution earlier on but now I've integrated the official 3D Slicer solution for the Python upgrade into my SlicerCAT app GeoSlicer. However I am using the latest stable version of Slicer instead of the master HEAD, hence sitk is on v2.0.2 and itk is on v5.1.2

**Comment by jcfr** [comment]
> SimpleITK_install_step_error.txt

@cpinter 

To help address this, you should be able to do the following:

```
set EP_EXECUTE_DISABLE_CAPTURE_OUTPUTS=1
cmake -P E:/t3/tR/SimpleITK_install_step.cmake
```

This should allow to reproduce the error.

After confirming you can reproduce the error, we need to understand what is missing.

Considering the following:
* Add "print()" calls to `E:/t3/tR/python-install/Lib/sitecustomize.py`
* Check the list of path configured in `E:/t3/tR/SimpleITK_Env.cmake`
* Did you build ITK against additional libraries that wouldn't be listed in  `SimpleITK_Env.cmake`

Currently, it is assumed the dependencies of SimpleITK will not be modified ... and these are hard-coded here: https://github.com/Slicer/Slicer/blob/015fa17a5715863052cda5dae2a14098589b01ae/SuperBuild/External_SimpleITK.cmake#L48-L85


These information should help you make progress

**Comment by jcfr** [comment]
@jamesobutler  

In this pull request, could you also consider updating the files setting the variable `ITK_NO_IO_FACTORY_REGISTER_MANAGER` ?

See https://github.com/Slicer/Slicer/search?q=ITK_NO_IO_FACTORY_REGISTER_MANAGER

As depicted [here](https://github.com/InsightSoftwareConsortium/ITK/blob/4f68349370abaaee7ee9799a19c4d99f3d914aec/CMake/ITKFactoryRegistration.cmake#L198) and https://github.com/InsightSoftwareConsortium/ITK/pull/3153, using `ITK_NO_IO_FACTORY_REGISTER_MANAGER` is now deprecated. 

Instead the following variables should be set:

```cmake
find_package(ITK ...)
if(ITK_VERSION VERSION_GREATER_EQUAL "5.3")
  foreach(factory_uc IN ITEMS "IMAGEIO" "MESHIO" "TRANSFORMIO")
    set(ITK_NO_${factory_uc}_FACTORY_REGISTER_MANAGER 1)
  endforeach()
else()
  set(ITK_NO_IO_FACTORY_REGISTER_MANAGER 1)
endif()
include(${ITK_USE_FILE})
```

While there are more improvement that could be done (e.g calling the new `itk_generate_factory_registration()` CMake macro instead of calling  `include(${ITK_USE_FILE})`), this approach allows to maintain backward compat for building against older ITK.

File: SuperBuild/External_SimpleITK.cmake [hunk 2]
```diff
@@ -177,7 +177,7 @@ ExternalProject_Execute(${proj} \"install\" \"${PYTHON_EXECUTABLE}\" setup.py in
       -DSimpleITK_PYTHON_THREADS:BOOL=ON
       -DSimpleITK_INSTALL_ARCHIVE_DIR:PATH=${Slicer_INSTALL_LIB_DIR}
       -DSimpleITK_INSTALL_LIBRARY_DIR:PATH=${Slicer_INSTALL_LIB_DIR}
-      -DSimpleITK_INT64_PIXELIDS:BOOL=OFF
+      -DSimpleITK_INT64_PIXELIDS:BOOL=ON
```
Comment (jcfr) [inline SuperBuild/External_SimpleITK.cmake:180]: Is there any commit or issue that could be referenced in the commit message to justify toggling this option ?

File: Base/Logic/vtkSlicerApplicationLogicRequests.h [hunk 3]
```diff
@@ -176,7 +176,7 @@ class ReadDataRequestFile : public DataRequest
         }
       else
         {
-        removed = itksys::SystemTools::RemoveFile(m_Filename.c_str());
+        removed = static_cast<bool>(itksys::SystemTools::RemoveFile(m_Filename.c_str()));
```
Comment (jcfr) [inline Base/Logic/vtkSlicerApplicationLogicRequests.h:179]: > Also includes changes where an explicit cast is required to convert itksys::Status into bool. Based on the same type changes made in https://github.com/InsightSoftwareConsortium/ITK/commit/b8ea8a11d47e95d4c03d24c9c4de36c35a3ee95d

This makes sense. I think this is a more elegant way of addressing this. For sake of consistency, I suggest to revisit the approach implemented in b3b16b31d144e578f336367e880e06961a0efa06 to deal with a similar change associated with `vtksys`

File: SuperBuild/External_SimpleITK.cmake (hunk 2 above)
Comment (jamesobutler) [reply SuperBuild/External_SimpleITK.cmake:180]: Sure I can add reference to https://github.com/SimpleITK/SimpleITK/commit/2a9b7cdc2fa87b75ccbc9b6eeda8c1ff1b08e77f

**Comment by ebrahimebrahim** [comment]
I tried to build this on linux, from 29ca6f9f7466b8a81008381aae155dbb4b145783, with `Slicer_BUILD_ITKPython` turned on. Just thought I'd let you to know that I ended up getting the following error:

```text
[ 98%] Built target ITKVideoCoreSwig
[ 98%] Linking CXX shared module ../../Generators/Python/itk/_ITKVideoCorePython.so
[ 98%] Built target ITKVideoCorePython
[ 98%] Generating ../../itkImageToVTKImageFilter.xml
In file included from /home/ebrahim/Slicer-SuperBuild-ButlerCPP14/ITK-build/Wrapping/itkImageToVTKImageFilter.cxx:16:
/home/ebrahim/Slicer-SuperBuild-ButlerCPP14/ITK/Modules/Bridge/VtkGlue/include/itkImageToVTKImageFilter.h:22:10: fatal error: 'vtkImageImport.h' file not found
#include "vtkImageImport.h"
         ^~~~~~~~~~~~~~~~~~
1 error generated.
make[5]: *** [Wrapping/Modules/ITKVtkGlue/CMakeFiles/ITKVtkGlueCastXML.dir/build.make:65: Wrapping/itkImageToVTKImageFilter.xml] Error 1
make[4]: *** [CMakeFiles/Makefile2:36718: Wrapping/Modules/ITKVtkGlue/CMakeFiles/ITKVtkGlueCastXML.dir/all] Error 2
make[3]: *** [Makefile:152: all] Error 2
make[2]: *** [CMakeFiles/ITK.dir/build.make:118: ITK-prefix/src/ITK-stamp/ITK-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:1487: CMakeFiles/ITK.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

**Comment by jamesobutler** [comment]
@ebrahimebrahim There is a patch proposed at https://discourse.slicer.org/t/addressing-issue-with-pip-install-itk/21763/3?u=jamesobutler which this PR isn't using yet for `Slicer_BUILD_ITKPython` support.

**Comment by ebrahimebrahim** [comment]
> There is a patch proposed

:thinking: I'm not so sure the patch is related to my error. In my error, it looks like VTK is not being found during part of the ITK build process. This seems like an issue to be fixed in the the Slicer superbuild system, which I am unfortunately not very good at manipulating.

**Comment by jcfr** [comment]
The patch is to avoid ITK built in the wheel from resolving global symbols exported by the itk built in Slicer. It will not address the error reported by @ebrahimebrahim

**Comment by jcfr** [comment]
Following integration of this pull-request, https://github.com/Slicer/Slicer/pull/6155 should also be integrated.

**Comment by cpinter** [comment]
@jcfr Thank you so much for the pointers! I just had time to look into it.

> Add "print()" calls to E:/t3/tR/python-install/Lib/sitecustomize.py

I added some prints to `slicer_dll_directories.py` under lines 7 and 14, `print(f'library_paths: {library_paths}')` and `print(f'os.add_dll_directory({path})')`, respectfully, and the output is:

```
E:\t3\tR>cmake -P E:/t3/tR/SimpleITK_install_step.cmake
-- SimpleITK: Removing 'install' log files
-- SimpleITK: SimpleITK_WORKING_DIR: E:/t3/tR/SimpleITK-build/SimpleITK-build/Wrapping/Python
-- SimpleITK: E:/t3/tR/python-install/bin/PythonSlicer.exe;setup.py;install
-- SimpleITK: 'install' Disabling capture of outputs (EP_EXECUTE_DISABLE_CAPTURE_OUTPUTS env. variable set to '1')
library_paths: ['E:\\t3\\tR\\python-install\\bin', 'E:\\t3\\tR\\OpenSSL-install\\Release\\bin', 'E:\\t3\\tR\\ITK-build\\bin\\Debug', 'E:\\t3\\tR\\ITK-build\\bin\\Release', 'E:\\t3\\tR\\ITK-build\\bin\\MinSizeRel', 'E:\\t3\\tR\\ITK-build\\bin\\RelWithDebInfo', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\Debug', 'E:\\t3\\tR\\DCMTK-build\\bin\\Debug', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\Release', 'E:\\t3\\tR\\DCMTK-build\\bin\\Release', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\MinSizeRel', 'E:\\t3\\tR\\DCMTK-build\\bin\\MinSizeRel', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\RelWithDebInfo', 'E:\\t3\\tR\\DCMTK-build\\bin\\RelWithDebInfo', 'E:\\t3\\tR\\tbb-install\\tbb2019_20191006oss\\bin\\intel64\\vc14', 'E:\\t3\\tR\\ITK-build\\bin\\Debug', 'E:\\t3\\tR\\ITK-build\\bin\\Release', 'E:\\t3\\tR\\ITK-build\\bin\\MinSizeRel', 'E:\\t3\\tR\\ITK-build\\bin\\RelWithDebInfo', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\Debug', 'E:\\t3\\tR\\DCMTK-build\\bin\\Debug', 'E:\\Traceback (most recent call last):
t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\Release', 'E:\\t3\\tR\\DCMTK-build\\bin\\Release', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\MinSizeRel', 'E:\\t3\\tR\\DCMTK-build\\bin\\MinSizeRel', 'E:\\t3\\tR\\SimpleITK-build\\SimpleITK-build\\bin\\RelWithDebInfo', 'E:\\t3\\tR\\DCMTK-build\\bin\\RelWithDebInfo', 'E:\\t3\\tR\\tbb-install\\tbb2019_20191006oss\\bin\\intel64\\vc14', 'E:\\Utils\\NVIDIA GPU Computing Toolkit\\v10.2\\bin', 'E:\\Utils\\NVIDIA GPU Computing Toolkit\\v10.2\\libnvvp', 'C:\\Windows\\system32', 'C:\\Windows', 'C:\\Windows\\System32\\Wbem', 'C:\\Windows\\System32\\WindowsPowerShell\\v1.0', 'C:\\Windows\\System32\\OpenSSH', 'C:\\Program Files\\NVIDIA Corporation\\NVIDIA NGX', 'C:\\Program Files\\NVIDIA Corporation\\NVIDIA NvDLISR', 'C:\\Program Files (x86)\\Intel\\Intel(R) Management Engine Components\\DAL', 'C:\\Program Files\\Intel\\Intel(R) Management Engine Components\\DAL', 'C:\\Program Files\\Git\\cmd', 'C:\\Program Files\\SlikSvn\\bin', 'C:\\Program Files\\PuTTY', 'C:\\Prog  File "E:\t3\tR\SimpleITK-build\SimpleITK-build\Wrapping\Python\setup.py", line 14, in <module>
    from SimpleITK._version import __version__
  File "E:\t3\tR\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK\__init__.py", line 18, in <module>
    from SimpleITK.SimpleITK import *
  File "E:\t3\tR\SimpleITK-build\SimpleITK-build\Wrapping\Python\SimpleITK\SimpleITK.py", line 13, in <module>
    from . import _SimpleITK
ImportError: DLL load failed while importing _SimpleITK: The specified procedure could not be found.
ram Files\\AutoFirma\\AutoFirma', 'C:\\WINDOWS\\system32', 'C:\\WINDOWS', 'C:\\WINDOWS\\System32\\Wbem', 'C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0', 'C:\\WINDOWS\\System32\\OpenSSH', 'C:\\Program Files\\TortoiseGit\\bin', 'C:\\Users\\pinte\\AppData\\Local\\Microsoft\\WindowsApps', 'e:\\Utils\\Microsoft VS Code\\bin', 'd:\\OneDrive\\Utils\\dcmtk-3.6.5-win64-dynamic\\bin', 'C:\\Program Files\\CMake\\bin', 'C:\\WinNMP\\bin', 'C:\\Users\\pinte\\AppData\\Local\\gitkraken\\bin']
os.add_dll_directory(E:\t3\tR\python-install\bin)
os.add_dll_directory(E:\t3\tR\OpenSSL-install\Release\bin)
os.add_dll_directory(E:\t3\tR\ITK-build\bin\Release)
os.add_dll_directory(E:\t3\tR\SimpleITK-build\SimpleITK-build\bin\Release)
os.add_dll_directory(E:\t3\tR\DCMTK-build\bin\Release)
os.add_dll_directory(E:\t3\tR\tbb-install\tbb2019_20191006oss\bin\intel64\vc14)
os.add_dll_directory(E:\t3\tR\ITK-build\bin\Release)
os.add_dll_directory(E:\t3\tR\SimpleITK-build\SimpleITK-build\bin\Release)
os.add_dll_directory(E:\t3\tR\DCMTK-build\bin\Release)
os.add_dll_directory(E:\t3\tR\tbb-install\tbb2019_20191006oss\bin\intel64\vc14)
os.add_dll_directory(E:\Utils\NVIDIA GPU Computing Toolkit\v10.2\bin)
os.add_dll_directory(E:\Utils\NVIDIA GPU Computing Toolkit\v10.2\libnvvp)
os.add_dll_directory(C:\Windows\system32)
os.add_dll_directory(C:\Windows)
os.add_dll_directory(C:\Windows\System32\Wbem)
os.add_dll_directory(C:\Windows\System32\WindowsPowerShell\v1.0)
os.add_dll_directory(C:\Windows\System32\OpenSSH)
os.add_dll_directory(C:\Program Files\NVIDIA Corporation\NVIDIA NGX)
os.add_dll_directory(C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR)
os.add_dll_directory(C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL)
os.add_dll_directory(C:\Program Files\Intel\Intel(R) Management Engine Components\DAL)
os.add_dll_directory(C:\Program Files\Git\cmd)
os.add_dll_directory(C:\Program Files\SlikSvn\bin)
os.add_dll_directory(C:\Program Files\PuTTY)
os.add_dll_directory(C:\Program Files\AutoFirma\AutoFirma)
os.add_dll_directory(C:\WINDOWS\system32)
os.add_dll_directory(C:\WINDOWS)
os.add_dll_directory(C:\WINDOWS\System32\Wbem)
os.add_dll_directory(C:\WINDOWS\System32\WindowsPowerShell\v1.0)
os.add_dll_directory(C:\WINDOWS\System32\OpenSSH)
os.add_dll_directory(C:\Program Files\TortoiseGit\bin)
os.add_dll_directory(C:\Users\pinte\AppData\Local\Microsoft\WindowsApps)
os.add_dll_directory(e:\Utils\Microsoft VS Code\bin)
os.add_dll_directory(d:\OneDrive\Utils\dcmtk-3.6.5-win64-dynamic\bin)
os.add_dll_directory(C:\Program Files\CMake\bin)
os.add_dll_directory(C:\WinNMP\bin)
os.add_dll_directory(C:\Users\pinte\AppData\Local\gitkraken\bin)
CMake Error at slicersources-src/CMake/ExternalProjectForNonCMakeProject.cmake:109 (message):
  SimpleITK: install step failed with exit code '1'.
Call Stack (most recent call first):
  SimpleITK_install_step.cmake:3 (ExternalProject_Execute)
```

> Check the list of path configured in E:/t3/tR/SimpleITK_Env.cmake

It seems normal. The content is exactly the same as in the build tree that I built successfully before updating the Slicer hash. It contains the correct paths (for all four configurations but I assume it's OK) of ITK, SimpleITK, DCMTK, and TBB.

> Did you build ITK against additional libraries that wouldn't be listed in SimpleITK_Env.cmake ?

No, I didn't include any non-default component in ITK if this is the question. It is configured by Slicer as usual, and `Slicer_ITK_ADDITIONAL_MODULES` is empty.

Not sure where to go from here, everything seems to be in order so far. If I had a way to figure out what is the procedure mentioned in the message `The specified procedure could not be found`, or even just which DLL failed loading...

**Comment by jamesobutler** [comment]
@cpinter You may want to open up a new Slicer github issue or open up a discourse post to continue this discussion and debugging of the issues you are facing. This being because the issue you are having is with current Slicer `master` and not necessarily with the changes to latest ITK and SimpleITK that this specific PR is introducing.

File: Modules/CLI/CastScalarVolume/CMakeLists.txt [hunk 4]
```diff
@@ -18,7 +18,13 @@ set(${PROJECT_NAME}_ITK_COMPONENTS
   ITKImageFilterBase
   )
 find_package(ITK 4.6 COMPONENTS ${${PROJECT_NAME}_ITK_COMPONENTS} REQUIRED)
-set(ITK_NO_IO_FACTORY_REGISTER_MANAGER 1) # See Libs/ITKFactoryRegistration/CMakeLists.txt
+if(ITK_VERSION VERSION_GREATER_EQUAL "5.3")
+  foreach(factory_uc IN ITEMS "IMAGEIO" "MESHIO" "TRANSFORMIO")
+    set(ITK_NO_${factory_uc}_FACTORY_REGISTER_MANAGER 1)
+  endforeach()
+else()
+  set(ITK_NO_IO_FACTORY_REGISTER_MANAGER 1)
```
Comment (jcfr) [inline Modules/CLI/CastScalarVolume/CMakeLists.txt:26]:
```suggestion
  set(ITK_NO_IO_FACTORY_REGISTER_MANAGER 1) # See Libs/ITKFactoryRegistration/CMakeLists.txt
```

Comment (jamesobutler) [reply Modules/CLI/CastScalarVolume/CMakeLists.txt]: I've restored this comment that I had removed.

**Comment by jcfr** [comment]
@jamesobutler I am starting a test build on all platform, once experimental results comes back. I suggest we integrate. How does that sound ?

**Comment by jamesobutler** [comment]
One of the reasons I had this as a draft was my expectation that this would likely get integrated after a new Slicer stable release. Have you all decided to move forward with this for including in the next Slicer stable? 

I was also waiting for ITK to provide an official tag for 5.3.0 as according to their milestone https://github.com/InsightSoftwareConsortium/ITK/milestone/17, it is due tomorrow. So could wait just a couple days to get a real tag instead of some git hash. You may have some insights if this is still on track or not.

**Comment by jamesobutler** [comment]
This being #5673 was put into the Slicer 5.1 milestone by @lassoan so my expectation is that this comes after a new Slicer stable is released. It’s been just about 1 year since the last one.

**Comment by dzenanz** [comment]
The official tag for 5.3.0 is not coming out tomorrow. It will probably be out in weeks, not days. I updated the milestone to reflect this.

**Comment by jcfr** [comment]
Experimental builds of this branch have been initiated  on the three platforms :hourglass: 

| image |
|--|
| https://slicer.cdash.org/index.php?project=SlicerPreview&date=2022-02-15&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=c%2B%2B14-itk-post-5.3rc03 |

**Comment by jamesobutler** [comment]
@jcfr The experimental builds have completed with testing results matching the same as the currently Slicer Preview builds so things look ready for integration.

**Pull request merged**